include/clasp/core/array_double.h

```diff
@@ -24,8 +24,8 @@ namespace core {
     typedef specialized_SimpleVector_double TemplatedBase;
   public:
     static value_type default_initial_element(void) {return 0.0;}
-    static value_type from_object(T_sp obj) { return clasp_to_double(obj);};
-    static T_sp to_object(const value_type& v) { return core::clasp_make_double_float(v); };
+    static value_type from_object(T_sp obj) { return clasp_to_double(gc::As<Number_sp>(obj));};
```

Comment:
This should use gc::As_unsafe<Number_sp>(obj). This would install a type check whenever from_object is called and we don't want them automatically installed everywhere.